tools/lint-docstring

```diff
@@ -0,0 +1,150 @@
+            padded_folder_cell, padded_file_cell, padded_desc_cell = pad_column_cells(
+                text_tuple, COLUMN_WIDTHS
+            )
+            new_line = (
+                f"| {padded_folder_cell}| {padded_file_cell}| {padded_desc_cell}|\n"
+            )
+            table_markdown.append(new_line)
+            folder_text = " "
+
+        # adding blank row at the end of every folder
+        table_markdown.append(BLANK_LINE)
+
+    # Folders that do not contain any files with docstrings are added separately to the file-overview
+    for skipped_folder_name in sorted(DOCSTRINGS_FOR_NO_FILE_FOLDERS):
+        skipped_text_tuple = (
+            skipped_folder_name,
+            " ",
+            DOCSTRINGS_FOR_NO_FILE_FOLDERS[skipped_folder_name],
+        )
+        padded_folder_cell, padded_file_cell, padded_desc_cell = pad_column_cells(
+            skipped_text_tuple, COLUMN_WIDTHS
+        )
+        new_line = f"| {padded_folder_cell}| {padded_file_cell}| {padded_desc_cell}|\n"
+        table_markdown.extend([new_line, BLANK_LINE])
+    doc_data[TABLE_OFFSET - 1 :] = table_markdown[:-1]
+    updated_data = "".join(doc_data)
+    with open(DEVELOPER_DOC, "w") as developer_file_write:
+        developer_file_write.write(updated_data)
+
+
+def generate_folder_file_docstrings_dict() -> Dict[str, Dict[str, str]]:
+    """
+    Returns a dictionary containing folder name which in turn
+    is a dictionary containing files and their respective descriptions
+    """
+    total_files = extract_folder_file_structure()
+    folder_file_docstring: Dict[str, Dict[str, str]] = {}
+    for folder, files in sorted(total_files.items()):
+        folder_file_docstring[str(folder)] = {}
+        for file in files:
+            imported_file = importlib.import_module(
+                f'{folder.replace("/",".")}.{file[:-3]}'
+            )
+            extracted_docstring = str(imported_file.__doc__)
+            docstring = extracted_docstring.strip().replace("\n", " ")
+            if len(docstring) >= COLUMN_WIDTHS[2] + 1:
+                raise ValueError(
+                    f"{file} has docstring longer than/equal to {int(COLUMN_WIDTHS[2])+1}"
+                )
+            folder_file_docstring[str(folder)][file] = docstring
+    return folder_file_docstring
+
+
+def extract_folder_file_structure() -> Dict[str, List[str]]:
+    """
+    Returns dictionary containing folders and respective python files within them
+    """
+    folders_and_files = {}
+    for path_to_folder in glob.glob(f"{ZULIPTERMINAL}/**/", recursive=True):
+        complete_directory_path = Path(path_to_folder)
+        relative_directory_path = complete_directory_path.relative_to(ROOT_DIRECTORY)
+        if str(relative_directory_path) not in DOCSTRINGS_FOR_NO_FILE_FOLDERS:
+            files_in_directory = [
+                Path(file_name).name
+                for file_name in glob.glob(f"{complete_directory_path}/*.py")
```

Comment:
Probably using `complete_directory_path.glob('*.py')` makes the code simpler here. You can then directly do `file.name`